http/src/test/java/com/hazelcast/jet/contrib/http/HttpSinkTest.java

```diff
@@ -0,0 +1,316 @@
+
+        // cleanup
+        cleanup(job, eventSource);
+    }
+
+    private EventSource receiveFromSse(String sseAddress, Collection<String> queue) {
+        EventHandler eventHandler = new EventHandler() {
+            @Override
+            public void onOpen() {
+            }
+
+            @Override
+            public void onClosed() {
+            }
+
+            @Override
+            public void onMessage(String event, MessageEvent messageEvent) {
+                queue.add(messageEvent.getData());
+            }
+
+            @Override
+            public void onComment(String comment) {
+            }
+
+            @Override
+            public void onError(Throwable t) {
+            }
+        };
+        OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder();
+        if (sseAddress.startsWith("https")) {
+            clientBuilder.sslSocketFactory(sslContextFn().get().getSocketFactory(), x509TrustManager());
+            clientBuilder.hostnameVerifier(new NoopHostnameVerifier());
+        }
+        EventSource.Builder builder = new EventSource.Builder(eventHandler, URI.create(sseAddress));
+        builder.client(clientBuilder.build());
+        EventSource eventSource = builder.build();
+        eventSource.start();
+        assertTrueEventually(() -> assertSame(OPEN, eventSource.getState()));
+        return eventSource;
+    }
+
+    private WebSocketChannel receiveFromWebSocket(String webSocketAddress, Collection<String> queue) throws Exception {
+        WebSocketChannel wsChannel = connectWithRetry(webSocketAddress);
+        wsChannel.getReceiveSetter().set(new AbstractReceiveListener() {
+            @Override
+            protected void onFullTextMessage(WebSocketChannel channel, BufferedTextMessage message) {
+                queue.add(message.getData());
+            }
+        });
+        wsChannel.resumeReceives();
+        return wsChannel;
+    }
+
+    private WebSocketChannel connectWithRetry(String webSocketAddress) throws Exception {
+        Xnio xnio = Xnio.getInstance(HttpSinkTest.class.getClassLoader());
+        for (int i = 0; i < 30; i++) {
+            try {
+                WebSocketClient.ConnectionBuilder builder = WebSocketClient
+                        .connectionBuilder(worker, buffer, URI.create(webSocketAddress));
+                if (webSocketAddress.startsWith("wss")) {
+                    builder.setSsl(new UndertowXnioSsl(xnio, OptionMap.EMPTY, sslContextFn().get()));
+                }
+                return builder.connect().get();
+            } catch (Exception e) {
+                logger.warning(e.getMessage());
+                sleepAtLeastMillis(100);
+            }
+        }
+        throw new AssertionError("Failed to connect to " + webSocketAddress);
+    }
+
+    private Job startJob(Sink<Object> sink) {
+        Pipeline p = Pipeline.create();
+        p.readFrom(HttpListenerSources.httpListener())
```

Comment:
I think these sink tests should be reworked so they use some simple test source, not the `httpListener()`

Reply:
I've changed it to use an IQueue source instead